### PR TITLE
Use Github Actions for manual build test

### LINKED ISSUE #884: Manual build tests are currently a no-op
#### Description:
Cmdstan manual build tests are currently done with Circle CI. There is no issue with Circle CI itself, the problem is how we use it. 

There are two separate issues:
- we rely on a docker image from Sean's personal docker account. This is a minor issue.
- The tests do nothing. They catch no errors. This is a major issue.

I propose we switch to Github Actions for this tests. That eliminates the need for using a docker image. Its really a simple test and running it in the native Github CI is simple.

#### Reproducible Steps:

Make an error in any cmdstan manual .tex files, open a PR and see it pass Circle CI tests.

See this commit: 
https://github.com/stan-dev/cmdstan/pull/883/commits/6351b31f593a5eccdde8438208719945687c0230

Click on the green tick next to the commit message and see that the Circle CI tests pass.

#### Current Output:
Tests pass even with error in Tex.

#### Expected Output:
Tests should not pass with errors in Tex.

#### Current Version:
v2.23.0

### PR DESCRIPTION
**Pull request by rok-cesnovar**
#### Submisison Checklist

- [x] Run tests: `./runCmdStanTests.py src/test`
- [x] Declare copyright holder and open-source license: see below

#### Summary:

Fixes #884 by removing Circle CI and replacing it with Github Actions. If this is approved we also need to remove the Circle CI webhook in the repo settings. Otherwise Circle CI test fails as this PR removes the test yml file.

How to review: 

- check https://github.com/stan-dev/cmdstan/pull/883/commits/6351b31f593a5eccdde8438208719945687c0230 that the Circle CI passed even with error
- see that Github actions failed on the same error as it should: https://github.com/stan-dev/cmdstan/runs/726799350
- check the Github Actions setup. Its really intuitive, so no experience needed there
- see that test pass if error removed and the manual PDF is built on success: https://github.com/stan-dev/cmdstan/actions/runs/121292187

#### Copyright and Licensing

Please list the copyright holder for the work you are submitting (this will be you or your assignee, such as a university or company): Rok Češnovar, Univ. of Ljubljana

By submitting this pull request, the copyright holder is agreeing to license the submitted work under the following licenses:
- Code: BSD 3-clause (https://opensource.org/licenses/BSD-3-Clause)
- Documentation: CC-BY 4.0 (https://creativecommons.org/licenses/by/4.0/)

### CODE REVIEW

**Comment by rok-cesnovar** [comment]
This is ready for review. I updated the top-level comment of this PR with all the info.

**Comment by rok-cesnovar** [comment]
@ahartikainen mind looking at this when you have a few minutes. You are my go-to Github Actions expert :)

File: .github/workflows/manual.yml [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+
+    - name: Build manual
+      run: make manual
+    
+    - name: Upload build result
+      uses: actions/upload-artifact@v2
+      if: success()
+      with:
+        name: manual_PDF
+        path: 'doc/cmdstan-guide-2.23.pdf'
```
Comment (ahartikainen) [inline .github/workflows/manual.yml:26]: I think wildcard could work here

`doc/cmdstan-guide-*.pdf`

Comment (rok-cesnovar) [reply .github/workflows/manual.yml]: Indeed. Tnx

**Comment by ahartikainen** [comment]
Btw, you can also upload pdf to release if you want to set up that

https://github.com/marketplace/actions/upload-files-to-a-github-release


```
- name: Upload build result to release
      uses: svenstaro/upload-release-action@v1-release
      if: success() && startsWith(github.ref, 'refs/tags/v')
      with:
        repo_token: ${{ secrets.GITHUB_TOKEN }}
        file: doc/cmdstan-guide-*.pdf
        tag: ${{ github.ref }}
        overwrite: true
        file_glob: true
```

**Comment by rok-cesnovar** [comment]
Thanks Ari! I think Nic manages the release from Jenkins so lets leave that for now.  But if we move creating releases to G.A. that will definitely be very handy.

@serban-nicusor-toptal can you please approve for @ahartikainen (I dont understand why Ari does not have approve permissions and who can give them to him).

Also Nic, do you have permission to remove Circle CI from cmdstan? Based on [this link](https://discuss.circleci.com/t/remove-circleci-from-github-repo/1144) it should be: 
`On GitHub, go to repo settings, Webhooks and Services, and delete the CircleCi webhook.`

**Comment by serban-nicusor-toptal** [comment]
Should be done, let me know if everything is working fine.

**Comment by rok-cesnovar** [comment]
Great, will try. Will push a whitespace change to this branch.

**Comment by rok-cesnovar** [comment]
Yes, seems that did it. Thanks. Please approve when you have the chance.

**Comment by rok-cesnovar** [comment]
@serban-nicusor-toptal you will have to approve also :) 

@mitzimorris can you please give Ari permissions to approve here? I am asking him to review CI/Python related stuff quite often and it seems silly to me he cant approve.

**Pull request merged**